libs/agno/agno/agent/agent.py

```diff
@@ -2917,6 +2927,71 @@ async def _ahandle_tool_call_updates_stream(
                 _t.requires_user_input = False
                 _t.answered = True
 
+    def _store_media(self, run_messages: RunMessages, run_response: RunOutput, model_response: ModelResponse):
+        """Store media from run_messages in run_response for persistence"""
+        from agno.utils.log import log_debug
+        
+        # Store input media from run_messages in run_response for persistence
+        if run_messages.images:
+            for image in run_messages.images:
+                try:
+                    if image.url:
+                        image_id = str(uuid4())
+                        image_artifact = ImageArtifact(id=image_id, url=image.url)
+                    elif image.content:
+                        image_id = str(uuid4())
+                        image_artifact = ImageArtifact(id=image_id, content=image.content)
+                    else:
+                        continue
+                    
+                    self._add_image(image_artifact, run_response)
```

Comment:
Don't we want to make a distinction between generated images and provided images, in terms of the `RunOutput`?

Reply:
> So say they send an image and the model analyses it, now RunOutput.images is populated. I'd find that weird? Or is that ok?
you are right, storing input image in `RunOutput` directly could be a confusing DX. i'll update